go/adbc/driver/snowflake/connection.go

```diff
@@ -85,82 +127,165 @@ func (c *connectionImpl) GetObjects(ctx context.Context, depth adbc.ObjectDepth,
 	defer conn.Close()
 
 	gQueryIDs, gQueryIDsCtx := errgroup.WithContext(ctx)
+
 	queryFile := queryTemplateGetObjectsAll
 	switch depth {
 	case adbc.ObjectDepthCatalogs:
-		if catalog == nil {
-			queryFile = queryTemplateGetObjectsTerseCatalogs
-			// if the catalog is null, show the terse databases
-			// which doesn't require a database context
-			gQueryIDs.Go(func() error {
-				return conn.Raw(func(driverConn any) error {
-					rows, err := driverConn.(driver.QueryerContext).QueryContext(gQueryIDsCtx, "SHOW TERSE DATABASES", nil)
-					if err != nil {
-						return err
-					}
+		queryFile = queryTemplateGetObjectsTerseCatalogs
+		gQueryIDs.Go(func() error {
+			return conn.Raw(func(driverConn any) (err error) {
+				query := "SHOW TERSE /* ADBC:getObjectsCatalogs */ DATABASES"
+				if catalog != nil && len(*catalog) > 0 && *catalog != "%" && *catalog != ".*" {
+					query += " LIKE '" + escapeSingleQuoteForLike(*catalog) + "'"
+				}
+				query += " IN ACCOUNT"
 
-					terseDbQueryID = rows.(gosnowflake.SnowflakeRows).GetQueryID()
-					return rows.Close()
-				})
+				terseDbQueryID, err = getQueryID(gQueryIDsCtx, query, driverConn)
+				return
 			})
-		} else {
-			queryFile = queryTemplateGetObjectsCatalogs
-		}
+		})
 	case adbc.ObjectDepthDBSchemas:
 		queryFile = queryTemplateGetObjectsDbSchemas
+		gQueryIDs.Go(func() error {
+			return conn.Raw(func(driverConn any) (err error) {
+				query := "SHOW TERSE /* ADBC:getObjectsDBSchemas */ SCHEMAS"
+				if dbSchema != nil && len(*dbSchema) > 0 && *dbSchema != "%" && *dbSchema != ".*" {
+					query += " LIKE '" + escapeSingleQuoteForLike(*dbSchema) + "'"
+				}
+				if catalog == nil || isWildcardStr(*catalog) {
+					query += " IN ACCOUNT"
+				} else {
+					query += " IN DATABASE \"" + quoteTblName(*catalog) + "\""
+				}
+
+				showSchemaQueryID, err = getQueryID(gQueryIDsCtx, query, driverConn)
+				return
+			})
+		})
+
+		gQueryIDs.Go(func() error {
+			return conn.Raw(func(driverConn any) (err error) {
+				query := "SHOW TERSE /* ADBC:getObjectsDBSchemas */ DATABASES"
+				if catalog != nil && len(*catalog) > 0 && *catalog != "%" && *catalog != ".*" {
+					query += " LIKE '" + escapeSingleQuoteForLike(*catalog) + "'"
+				}
+				query += " IN ACCOUNT"
+
+				terseDbQueryID, err = getQueryID(gQueryIDsCtx, query, driverConn)
+				return
+			})
```

Comment:
Some of these are repeated across cases. Could we extract them out of the switch-case to avoid the duplication?